src/api/services/ingestion.go

```diff
@@ -274,28 +274,69 @@ func (i *IngestionService) UploadVcfGzToDrs(cfg *models.Config, drsBridgeDirecto
 	data := fmt.Sprintf("{\"path\": \"%s/%s\"}", drsBridgeDirectory, gzippedFileName)
 
 	var (
-		drsResp *http.Response
-		drsErr  error
+		drsId           string
+		drsResp         *http.Response
+		drsErr          error
+		attemptCount    int = 0
+		maxAttempts     int = 5
+		waitTimeSeconds int = 3
 	)
 	for {
+		// prepare upload request to drs
 		r, _ := http.NewRequest("POST", drsUrl+"/private/ingest", bytes.NewBufferString(data))
-		r.SetBasicAuth(drsUsername, drsPassword)
 
+		r.SetBasicAuth(drsUsername, drsPassword)
 		r.Header.Add("Content-Type", "application/json")
+
 		client := &http.Client{}
+
+		// perform request
 		drsResp, drsErr = client.Do(r)
+
+		// check for errors, possibly try again
 		if drsErr != nil {
 			fmt.Printf("Upload to DRS error: %s\n", drsErr)
-			return ""
-		} else {
-			fmt.Printf("Upload to DRS succeeded: %d\n", drsResp.StatusCode)
+
+			if attemptCount < maxAttempts {
+				// increment attempt counter
+				attemptCount++
+
+				// give it a few seconds break
+				time.Sleep(time.Duration(waitTimeSeconds * int(time.Second)))
+
+				fmt.Printf("trying again...\n")
+				continue
+			} else {
+				fmt.Printf("exiting upload loop...\n")
+				return "" // empty drs-id string
+			}
```

Comment:
Wow! This is so much better! It's a really well thought addition

If I may suggest, in the else block, we might want to make the message clearer: the upload has failed at every attempt.